cli_tools/gce_vm_image_import/importer/bootable_disk_processor.go

```diff
@@ -16,37 +16,103 @@ package importer
 
 import (
 	"context"
+	"fmt"
+	"log"
 	"os"
 	"path"
 	"strconv"
 	"strings"
 
-	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
-
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/disk"
 	daisy_utils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/daisycommon"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	daisyCompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"google.golang.org/api/compute/v1"
 )
 
 type bootableDiskProcessor struct {
-	workflow        *daisy.Workflow
-	userLabels      map[string]string
-	storageLocation string
-	uefiCompatible  bool
-	noExternalIP    bool
-	network         string
-	OS              string
+	ImportArguments
+	computeClient daisyCompute.Client
```

Comment:
Since this is such a huge interface, it makes understanding the intent of the calling code difficult. To solve that, specify the method that you're using, similar to below. This helps the reader understand which method you're using, and makes testing easier since you can create a mock without a framework.

https://github.com/GoogleCloudPlatform/compute-image-tools/blob/514ebce5f1ed4baaa8769e5a590367006c8d20a5/cli_tools/gce_vm_image_import/importer/importer.go#L187-L186

Reply:
Personally, I found using limited interfaces like this annoying: whenever I needed to use a new operation, I had to change the interface. Understanding the intent of the code is not really easier: it's obvious from the calling code what is the intent even though there are other, not used functions in the big interface. 

Testing might be easier without a framework, but with a framework it's the same.

Reply:
I start by reading the interface, so having well-defined dependencies helps me. But it's a good learning that we're not all like that :)

In this particular case, here's my guidance:
1. Implement a processor specifically for UEFI (there's a longer comment about that below)
2. Given that the new processor will only need the CreateDisk method, I'd strongly recommend using implicit interfaces to get the benefits that we've talked about. That said, if you and Zoran aren't fans, I'm fine to move forward with the consensus

Reply:
It sounds like Tony favors the large interface, so I'm fine with proceeding with that.